legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-snowflake/src/main/java/org/finos/legend/engine/persistence/components/relational/snowflake/SnowflakeSink.java

```diff
@@ -568,6 +572,35 @@ public IngestorResult performBulkLoad(Datasets datasets, Executor<SqlGen, Tabula
         return result;
     }
 
+    private void enhanceWithLoadStatistics(Executor<SqlGen, TabularData, SqlPlan> executor, Map<StatisticName, Object> stats, String queryId)
+    {
+        try
+        {
+            RelationalTransformer transformer = new RelationalTransformer(SnowflakeSink.get());
+            SqlPlan physicalPlanForQueryOperatorStats = transformer.generatePhysicalPlan(QueryStatsLogicalPlanUtils.getLogicalPlanForQueryOperatorStats());
+            HashMap<String, PlaceholderValue> queryIdPlaceHolder = new HashMap<>();
+            queryIdPlaceHolder.put(QueryStatsLogicalPlanUtils.QUERY_ID_PARAMETER, PlaceholderValue.builder().value(queryId).isSensitive(false).build());
+            List<TabularData> queryOperatorStats = executor.executePhysicalPlanAndGetResults(physicalPlanForQueryOperatorStats, queryIdPlaceHolder);
+            if (!queryOperatorStats.isEmpty())
+            {
+                List<Map<String, Object>> queryOperatorStatsResults = queryOperatorStats.get(0).data();
+                queryOperatorStatsResults.forEach(queryStats -> {
+                    switch ((String) queryStats.get(QueryStatsLogicalPlanUtils.OPERATOR_TYPE_ALIAS))
+                    {
+                        case QueryStatsLogicalPlanUtils.EXTERNAL_SCAN_STAGE:
+                            stats.put(StatisticName.EXTERNAL_BYTES_SCANNED, queryStats.get(QueryStatsLogicalPlanUtils.EXTERNAL_BYTES_SCANNED_ALIAS));
+                        case QueryStatsLogicalPlanUtils.INSERT_STAGE:
+                            stats.put(StatisticName.INCOMING_RECORD_COUNT, queryStats.get(QueryStatsLogicalPlanUtils.INPUT_ROWS_ALIAS));
+                    }
+                });
+            }
+        }
+        catch (Exception e)
+        {
+            LOGGER.error(String.format("Error extracting query stats for query id: [%s].", queryId), e);
```

Comment:
Just checking : the idea to not throw the exception here is -- if we get an exception here, we should let the load still succeed!